src/libs/ReportUtils.ts

```diff
@@ -831,6 +836,40 @@ function isConciergeChatReport(report: OnyxEntry<Report>): boolean {
     return report?.participantAccountIDs?.length === 1 && Number(report.participantAccountIDs?.[0]) === CONST.ACCOUNT_ID.CONCIERGE && !isChatThread(report);
 }
 
+/**
+ * Checks if the supplied report belongs to workspace based on the provided params.
+ */
+function doesReportBelongToWorkspace(report: Report, policyID: string, policyMembersAccountIDs: string[]) {
+    return (
+        isConciergeChatReport(report) ||
+        (report.policyID === CONST.POLICY.ID_FAKE || !report.policyID ? hasReportCommonPolicyMemberWithArray(report, policyMembersAccountIDs) : report.policyID === policyID)
+    );
+}
+
+/**
+ * Given a collection of reports, return them sorted by the last read timestamp. Filters the sorted reports by a policy ID and policyMemberAccountIDs, if provided.
+ */
+function sortReportsByLastRead(
+    reports: OnyxCollection<Report>,
+    reportMetadata: OnyxCollection<ReportMetadata>,
+    policyID?: string,
+    policyMemberAccountIDs: string[] = [],
+): Array<OnyxEntry<Report>> {
+    let reportsValues = Object.values(reports ?? {});
+    if (policyID) {
+        reportsValues = reportsValues.filter((report) => !!report && doesReportBelongToWorkspace(report, policyID, policyMemberAccountIDs));
+    }
```

Comment:
Should we move this before the `sortReportsByLastRead ` method/ outside? This is now changing the purpose of the method which was only to sort so I fee like any filtering should be done before that and we only pass the collection of the reports we know we want to sort

Reply:
I've created the separated method to filter reports which is invoked before the `sortReportsByLastRead`, but to achieve that I had to move parsing OnyxCollection to the array to the parent function. Is it ok? This change required adjusting types in the function params.